code/modules/hydroponics/gene_modder.dm

```diff
@@ -244,6 +266,40 @@
 
 	data["modal"] = ui_modal_data(src)
 
+	var/list/stats = list()
+	var/list/traits = list()
+	var/list/reagents = list()
+	var/empty_disks = 0
+
+	data["stat_disks"] = list()
+	data["trait_disks"] = list()
+	data["reagent_disks"] = list()
+
+	for(var/i in 1 to length(contents))
+		if(istype(contents[i], /obj/item/disk/plantgene))
+			var/obj/item/disk/plantgene/D = contents[i]
+			if(!D.gene && !D.is_bulk_core)
+				empty_disks++
+			else if(D.is_bulk_core)
+				stats.Add(list(list("display_name" = D.ui_name, "index" = i, "stat" = "All", "ready" = D.seeds_needed <= D.seeds_scanned, "read_only" = D.read_only)))
+			else if(istype(D.gene, /datum/plant_gene/core))
+				var/datum/plant_gene/core/C = D.gene
+				stats.Add(list(list("display_name" = C.name +" "+ num2text(C.value), "index" = i, "stat" = C.name, "read_only" = D.read_only)))
+			else if(istype(D.gene, /datum/plant_gene/trait))
+				var/insertable = D.gene?.can_add(seed)
+				traits.Add(list(list("display_name" = D.gene.name, "index" = i, "can_insert" = insertable, "read_only" = D.read_only)))
+			else if(istype(D.gene, /datum/plant_gene/reagent))
+				var/datum/plant_gene/reagent/R = D.gene
+				var/insertable = R?.can_add(seed)
+				reagents.Add(list(list("display_name" = R.name + " " + num2text(R.rate*100) +"%", "index" = i,  "can_insert" = insertable, "read_only" = D.read_only)))
```

Comment:
Could be converted into a string like this:
`"display_name" = "[R.name] [num2text(R.rate*100)]%"`

makes it more readable.

Reply:
done